vault/resource_consul_secret_backend.go

```diff
@@ -288,3 +296,48 @@ func consulSecretBackendExists(d *schema.ResourceData, meta interface{}) (bool,
 func consulSecretBackendConfigPath(backend string) string {
 	return strings.Trim(backend, "/") + "/config/access"
 }
+
+func consulSecretsBackendCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+	oldToken, newToken := diff.GetChange("token")
+	ob, nb := diff.GetChange("bootstrap")
+	oldBootstrap := ob.(bool)
+	newBootstrap := nb.(bool)
+
+	// Disallow the following bad states on existing resource updates
+	if oldBootstrap && oldToken == "" {
+		// If the user already has a bootstrap resource created, but tries adding a token
+		if newToken != "" {
+			return fmt.Errorf("cannot change field `token` on a Consul secret backend resource with bootstrap")
```

Comment:
these errors seem like great candidates for some table driven unit tests to capture all the possible scenarios

Reply:
Good call out 👍